src/NHapi.Base/Parser/DefaultModelClassFactory.cs

```diff
@@ -1,277 +1,287 @@
-		/// the corresponding messages are parsed).  For each HL7 version you want to support, you must 
-		/// put a text file on your class path, under the folder /custom_packages, named after the version.  For example, 
-		/// for version 2.4, you might put the file "custom_packages/2.4" in your application JAR.  Each line in the  
-		/// file should name a package to search for message classes of that version.  For example, if you 
-		/// work at foo.org, you might create a v2.4 message structure called "ZFO" and define it in the class
-		/// <code>org.foo.hl7.custom.message.ZFO</code>.
-		/// </para>
-		/// <para>
-		/// In order for parsers to find this message
-		/// class, you would need to enter the following line in custom_packages/2.4: org.foo.hl7.custom
-		/// </para>
-		/// <para>
-		/// Packages are searched in the order specified.  The standard package for a given version
-		/// is searched last, allowing you to override the default implementation.  Please note that 
-		/// if you create custom classes for messages, segments, etc., their names must correspond exactly 
-		/// to their names in the message text.  For example, if you subclass the QBP segment in order to 
-		/// add your own fields, your subclass must also be called QBP. although it will obviously be in 
-		/// a different package.  To make sure your class is used instead of the default implementation, 
-		/// put your package in the package list.  User-defined packages are searched first, so yours 
-		/// will be found first and used.
-		/// </para>
-		/// <para>
-		/// It is important to note that there can only be one implementation of a particular message 
-		/// structure (i.e. one class with the message structure name, regardless of its package) among 
-		/// the packages defined as per the <code>PackageList()</code> method.  If there are duplicates 
-		/// (e.g. two ADT_A01 classes) the first one in the search order will always be used.  However, 
-		/// this restriction only applies to message classes, not segment classes, etc.  This is because 
-		/// classes representing parts of a message are referenced explicitly in the code for the message 
-		/// class, rather than being looked up (using findMessageClass() ) based on the String value of MSH-9.
-		/// </para>  
-		/// </summary>
-		public static List<string> PackageList(String version)
-		{
-			//load package lists if necessary ... 
+            return mc;
+        }
 
-			if (packages == null)
-			{
-				lock (_lockObject)
-				{
-					if (packages == null)
-					{
-						_isLoadingPackages = true;
-						IList<Hl7Package> packageList = PackageManager.Instance.GetAllPackages();
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="theName"></param>
+        /// <param name="theVersion"></param>
+        /// <returns></returns>
+        public virtual Type GetGroupClass(string theName, string theVersion)
```

Comment:
do we want expression bodies for methods? if so we can change this and the below?

Reply:
see related comment